### ISSUE TITLE
Nested comma triggers signature help

### ISSUE THREAD

**Issue by JsonFreeman** [Microsoft/TypeScript#12045]
**TypeScript Version:**  2.0.3

**Code**

```ts
function foo(x) { }
foo([0 ]) // Type a comma after the 0
```

**Expected behavior:**
In VS code, typing the comma should not trigger signature help
**Actual behavior:** 
Comma triggers signature help

**Comment by RyanCavanaugh** [comment]
I don't think it *should* trigger signature help -- it would make it look like you were outside the array. Thoughts?

**Comment by JsonFreeman** [comment]
It also happens in object literals and in nested functions.

**Comment by DanielRosenwasser** [comment]
Come on Jason, you don't remember this issue from ~2 years ago? #1150.

**Comment by JsonFreeman** [comment]
Ah, it's all coming back to me now. I like the TriggerReason idea. Could that be implemented for VSCode too?

**Comment by mhegazy** [comment]
indeed a duplicate of #1150.
